native/src/widget/text_input.rs

```diff
@@ -792,6 +792,11 @@ impl State {
     pub fn move_cursor_to(&mut self, position: usize) {
         self.cursor.move_to(position);
     }
+
+    /// Selects all the content of the [`TextInput`].
+    pub fn select_all(&mut self) {
+        self.cursor.select_range(0, usize::MAX);
+    }
```

Comment:
I think it would make more sense to expose a more general `select` function, accepting both the start and the end, then in case you needed to select all you would use it just like you currently call `cursor.select_range`.

I'm not sure if there's a way to clip the range to the actual text size, since the `Value` is not available for the `State`, but right now this is done anyway on the Renderer: https://github.com/hecrj/iced/blob/f076649fbb575ee036ab0b3f4511690e3379c115/graphics/src/widget/text_input.rs#L157-L158

Reply:
I think `select_all` satisfies a clear use case.

We can expose a `select_range` or similar and focus on those use cases in a different unit of work.